resolver/src/lookup.rs

```diff
@@ -62,7 +81,14 @@ impl Lookup {
         rdatas.extend_from_slice(&*self.rdatas);
         rdatas.extend_from_slice(&*other.rdatas);
 
-        Self::new(Arc::new(rdatas))
+        // If both lookups have TTLs, choose the lower one.
+        match (self.ttl(), other.ttl()) {
```

Comment:
It seems like we need to compare the expiration times, not the TTLs, here.

Reply:
It depends on what you want to track here. The raw TTL, or the time in which the the record is no longer valid. The RFCs definitely recommend the latter.